core/services/feeds/service_test.go

```diff
@@ -5254,6 +5238,177 @@ func logMessages(logEntries []observer.LoggedEntry) []string {
 	return messages
 }
 
+func Test_Service_GetJobRuns(t *testing.T) {
+	t.Parallel()
+
+	var (
+		remoteUUID     = uuid.New()
+		feedsManagerID = int64(1)
+		jobID          = int32(100)
+		args           = &feeds.GetJobRunsArgs{
+			FeedsManagerID: feedsManagerID,
+			RemoteUUID:     remoteUUID,
+			Limit:          10,
+		}
+
+		testJob = job.Job{
+			ID:            jobID,
+			ExternalJobID: remoteUUID,
+			Name:          null.StringFrom("test job"),
+			Type:          "directrequest",
+		}
+
+		run1 = pipeline.Run{
+			ID:          1,
+			State:       pipeline.RunStatusCompleted,
+			CreatedAt:   time.Now().Add(-2 * time.Hour),
+			FinishedAt:  null.TimeFrom(time.Now().Add(-1 * time.Hour)),
+			AllErrors:   pipeline.RunErrors{null.StringFrom("test error"), null.String{}},
+			FatalErrors: pipeline.RunErrors{null.String{}, null.String{}},
+		}
+
+		run2 = pipeline.Run{
+			ID:          2,
+			State:       pipeline.RunStatusErrored,
+			CreatedAt:   time.Now().Add(-1 * time.Hour),
+			FinishedAt:  null.Time{},
+			AllErrors:   pipeline.RunErrors{null.String{}, null.StringFrom("fatal error")},
+			FatalErrors: pipeline.RunErrors{null.String{}, null.StringFrom("fatal error")},
+		}
+
+		run3 = pipeline.Run{
+			ID:          3,
+			State:       pipeline.RunStatusCompleted,
+			CreatedAt:   time.Now().Add(-30 * time.Minute),
+			FinishedAt:  null.TimeFrom(time.Now().Add(-10 * time.Minute)),
+			AllErrors:   pipeline.RunErrors{null.String{}, null.String{}},
+			FatalErrors: pipeline.RunErrors{null.String{}, null.String{}},
+		}
+	)
+
+	testCases := []struct {
+		name    string
+		args    *feeds.GetJobRunsArgs
+		before  func(svc *TestService)
+		want    int
+		wantErr string
+	}{
+		{
+			name: "success - returns job runs",
+			args: args,
+			before: func(svc *TestService) {
+				svc.jobORM.On("FindJobByExternalJobID", mock.Anything, remoteUUID).Return(testJob, nil)
```

Comment:
```suggestion
				svc.jobORM.EXPECT().FindJobByExternalJobID( mock.Anything, remoteUUID).Return(testJob, nil)
```

The `On("string")` api from testify is considered deprecated, as it's more error prone. I've added a few suggestions in the PR but will stop now to shorten the review.